watchdog.go

```diff
@@ -164,18 +183,13 @@ func watch() {
 			eventIsGc = true
 			// exit select.
 
-		case <-watchdog.closing:
-			runtime.SetFinalizer(&sentinelObj, nil) // clear the sentinel finalizer.
-
-			lk.Lock()
-			ch := gcTriggered
-			gcTriggered = nil
-			lk.Unlock()
-
+		case <-_watchdog.closing:
 			// close and drain
-			close(ch)
-			for range ch {
+			close(gcTriggered)
```

Comment:
This can race with a write to this channel.

Reply:
I don't think so: https://github.com/raulk/go-watchdog/pull/1/files#diff-fd8c8dd9c4dd131563030cfdc059a6e45c19a1173bd49d0885ddf019858f6b86R151-R155.

Reply:
The scheduler can pause here: https://github.com/raulk/go-watchdog/pull/1/files#diff-fd8c8dd9c4dd131563030cfdc059a6e45c19a1173bd49d0885ddf019858f6b86R163